src/V3Increments.cpp

```diff
@@ -0,0 +1,204 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//*************************************************************************
+// DESCRIPTION: Verilator: Replace return/continue with jumps
+//
+// Code available from: https://verilator.org
+//
+//*************************************************************************
+//
+// Copyright 2003-2020 by Wilson Snyder. This program is free software; you
+// can redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License
+// Version 2.0.
+// SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+//
+//*************************************************************************
+// V3Increments's Transformations:
+//
+// XXX: TODO: Write some docs
+//
+//*************************************************************************
+
+#include "config_build.h"
+#include "verilatedos.h"
+
+#include "V3Global.h"
+#include "V3Increments.h"
+#include "V3Ast.h"
+
+#include <algorithm>
+#include <cstdarg>
+#include <vector>
+
+#include <iostream>
+
+//######################################################################
+
+class IncrementsVisitor : public AstNVisitor {
+private:
+    // STATE
+    int                 m_modIncrementsNum;  // Var name counter
+
+    // METHODS
+    void insertBefore(AstNode* placep, AstNode* newp) {
+        AstNRelinker linker;
+        placep->unlinkFrBack(&linker);
+        newp->addNext(placep);
+        linker.relink(newp);
+    }
+
+public:
+    // CONSTRUCTORS
+    explicit IncrementsVisitor(AstNetlist* nodep) {
+        m_modIncrementsNum = 0;
+        iterate(nodep);
+    }
+
+    virtual ~IncrementsVisitor() {}
+
+    virtual void visit(AstPreInc* nodep) VL_OVERRIDE {
+        iterateChildren(nodep);
+
+        AstNodeVarRef *vr = VN_CAST(nodep->op1p(), VarRef);
+        AstNode *back = nodep->backp();
```

Comment:
"p" suffix on pointer variables

```suggestion
        AstNodeVarRef *varrefp = VN_CAST(nodep->op1p(), VarRef);
        AstNode *backp = nodep->backp();
```